ddtrace/tracer.py

```diff
@@ -98,7 +91,12 @@ def __init__(self, url=DEFAULT_AGENT_URL, dogstatsd_url=DEFAULT_DOGSTATSD_URL):
         hostname = self.DEFAULT_HOSTNAME
         port = self.DEFAULT_PORT
         writer = None
-        if url is not None:
+
+        if Tracer._is_agentless_environment() and url is None:
+            writer = LogWriter()
+        else:
+            if url is None:
+                url = self.DEFAULT_AGENT_URL
```

Comment:
I think this whole block can be simplified:
```python
if url is None:
     url = self.DEFAULT_AGENT_URL
     if self._agentless_environment():
        writer = LogWriter()
```

Reply:
I'm not sure that's doing the same thing. How do we determine to set the AgentWriter?